sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,164 +1,122 @@
-This will allow us to add things like `IgnoreAdditionalProperties` and `Version` to the options without needing to have our own ModelSerializer.
-The `SerializableOptions` would become internal and we would have a converter to convert from `JsonSerializerOptions` + `ModelJsonConverter` to `SerializableOptions`.
-
-### Serialization
-
-```C# Snippet:ModelConverter_Serialize
-DogListProperty dog = new DogListProperty
+```C# Snippet:CastOperations
+DefaultAzureCredential credential = new DefaultAzureCredential();
+PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), credential);
+Dog dog = new Dog
 {
     Name = "Doggo",
-    IsHungry = true,
-    Weight = 1.1,
-    FoodConsumed = { "kibble", "egg", "peanut butter" },
+    Age = 7
 };
 
-JsonSerializerOptions options = new JsonSerializerOptions();
-options.Converters.Add(new ModelJsonConverter());
+// Our models contain an implicit cast to RequestContent so you can pass them directly to protocol methods.
+Response response = client.CreatePet("myPet", dog);
 
-string json = System.Text.Json.JsonSerializer.Serialize(dog, options);
+// Our models also contain an explicit cast to the response type so you can deserialize them easily.
+dog = (Dog)response;
 ```
 
-### Deserialization
+## Using JsonSerializer
+
+If you would like to use the `JsonSerializer`, you can add the `ModelJsonConverter` to your options, which allow `JsonSerializer` to serialize and deserialize any model that implements `IModelJsonSerializable<T>`. The following example shows a Dog model being serialized and deserialized using the ModelJsonConverter.
 
-```C# Snippet:ModelConverter_Deserialize
-string json = @"[{""LatinName"":""Animalia"",""Weight"":1.1,""Name"":""Doggo"",""IsHungry"":false,""FoodConsumed"":[""kibble"",""egg"",""peanut butter""],""NumberOfLegs"":4}]";
+```C# Snippet:BaseModelConverter
+Dog dog = new Dog
+{
+    Name = "Doggo",
+    Age = 7
+};
 
 JsonSerializerOptions options = new JsonSerializerOptions();
+// The ModelJsonConverter is able to serialize and deserialize any model that implements IModelJsonSerializable<T>.
 options.Converters.Add(new ModelJsonConverter());
 
-DogListProperty dog = System.Text.Json.JsonSerializer.Deserialize<DogListProperty>(json, options);
+string json = JsonSerializer.Serialize(dog, options);
+
+dog = JsonSerializer.Deserialize<Dog>(json, options);
 ```
 
-## Envelope BYOM Case
+## Envelope bring your own model case
 
-The following examples show a use case where a User brings a model unknown to the Serializer. The serialization used for each model can also be set in the SerializableOptions options property Serializers. 
+The following examples show a use case where a user brings a model unknown to the Serializer. The serialization used for each model can also be set in the `ModelSerializableOptions` options property `GenericTypeSerializerCreator`. 
```

Comment:
what are models unknown to the serialize? Do you mean models that don't implement the interface? If yes, I would just say that.